PWGHF/Tasks/taskMcEfficiency.cxx

```diff
@@ -0,0 +1,290 @@
+
+        // all candidates
+        hCandidates->Fill(kHFStepTracked, candidate.pt(), mass, pdgCode, candidate.cpa(), collisionMatched);
+
+        // check if prongs have passed track cuts
+        if (checkTrack(candidate.prong0_as<TracksType>()) && checkTrack(candidate.prong1_as<TracksType>())) {
+          hCandidates->Fill(kHFStepTrackedCuts, candidate.pt(), mass, pdgCode, candidate.cpa(), collisionMatched);
+        }
+
+        if (!selected) {
+          continue;
+        }
+
+        // selected candidates
+        hCandidates->Fill(kHFStepTrackedSelected, candidate.pt(), mass, pdgCode, candidate.cpa(), collisionMatched);
+
+        // duplicates
+        int64_t hash = 0;
+        if (candidate.prong0Id() < candidate.prong1Id()) {
+          hash = ((int64_t)candidate.prong0Id() << 32) | candidate.prong1Id();
+        } else {
+          hash = ((int64_t)candidate.prong1Id() << 32) | candidate.prong0Id();
+        }
+        if (duplicates.find(hash) != duplicates.end()) {
+          hCandidates->Fill(kHFStepTrackedDuplicates, candidate.pt(), mass, pdgCode, candidate.cpa(), collisionMatched);
+        }
+        duplicates[hash]++;
+      }
+
+      auto hDuplicateCount = registry.get<TH1>(HIST("hDuplicateCount"));
+      for (const auto& i : duplicates) {
+        hDuplicateCount->Fill(i.second);
+      }
+    }
+  }
+
+  void processData(soa::Join<aod::HfCand2Prong, aod::HfSelD0>& candidates, TracksWithSelection& tracks)
+  {
+    candidateLoop<false>(candidates, tracks, tracks); // NOTE third argument has to be provided but is not used as template argument is <false>
+  }
+  PROCESS_SWITCH(HfTaskMcEfficiency, processData, "Process data (no MC information needed)", false);
+
+  void processMc(soa::Join<aod::HfCand2Prong, aod::HfSelD0>& candidates, TracksWithSelectionMC& tracks, aod::McParticles& mcParticles, aod::McCollisionLabels&)
+  {
+    candidateLoop<true>(candidates, tracks, mcParticles);
+
+    auto hCandidates = registry.get<StepTHn>(HIST("hCandidates"));
+    auto hTrackablePtEta = registry.get<StepTHn>(HIST("hTrackablePtEta"));
+
+    // lists for optimization
+    std::vector<bool> tracked(mcParticles.size(), false);
+    std::vector<bool> hasITS(mcParticles.size(), false);
+    std::vector<bool> hasTPC(mcParticles.size(), false);
+    std::vector<bool> selected(mcParticles.size(), false);
+    for (auto& track : tracks) {
+      if (track.mcParticleId() >= 0) {
+        tracked[track.mcParticleId()] = true;
+        if (checkTrack(track)) {
+          selected[track.mcParticleId()] = true;
+        }
+        if (track.hasITS()) {
+          hasITS[track.mcParticleId()] = true;
+        }
+        if (track.hasTPC()) {
+          hasTPC[track.mcParticleId()] = true;
+        }
+      }
+    }
+
+    for (auto pdgCode : (std::vector<int>)pdgCodes) {
```

Comment:
```suggestion
    for (const auto pdgCode : (std::vector<int>)pdgCodes) {
```

Reply:
Wouldn't just `pdgCodes.value` work instead of `(std::vector<int>)pdgCodes`?

Reply:
Yes!